### ISSUE TITLE
[Bug]: `isLoading$` never emits `false` when using the `AutoLoginAllRoutesGuard`

### ISSUE THREAD

**Issue by FabienDehopre** [damienbod/angular-auth-oidc-client#1493]
**What Version of the library are you using?**
14.1.2

**Describe the bug**
If you use the router guard `AutoLoginAllRoutesGuard`, the `isLoading$` never emits `false` because the guard is not using the main `OidcSecurityService` where the observable is changed from `true` to `false`.

**To Reproduce**

1. Add the following code in the AppComponent in the sample-code-flow-auto-login-all-routes sample app:
```ts
constructor(oidcSecurityService: OidcSecurityService) {
    oidcSecurityService.isLoading$.subscribe((isLoading) => {
      console.log('ISLOADING', isLoading);
    });
  }
```
2. Launch the sample project
3. Connect using e-mail and password
4. Look in the console for the string 'ISLOADING' => it will never display `false`

**Expected behavior**
I would expect that the `isLoading$` observable emits `false` once the config is loaded and the service is ready to be used even if it's the router guard who is doing the `checkAuth` call.

**Desktop (please complete the following information):**

- OS: Windows 10
- Browser Chrome
- Version 103

**Comment by FabianGosebrink** [comment]
The problem here is that we have logic in the facade, which is not good and was not intended to be there. The logic is only triggered, like you said, when using the facade. So we have to get rid of the logic in the facade and move it either into a service or make the `CheckAuthService` stateful. However, the best solution would be to use the existing `PublicEventsService` for this as this is what he was made for. But this would be a breaking change. Thoughts? @damienbod ?

**Comment by FabienDehopre** [comment]
Indeed, `PublicEventsService` will do the trick. 
Would be a good idea to either deprecate the `isLoading$` property or to rework it because it's a bit confusing.

**Comment by FabianGosebrink** [comment]
The `isLoading$` will be removed in future versions then.

**Comment by FabianGosebrink** [comment]
While fixing/extending this, I found out that the error case in case there is an error when checking the auth never worked. So it had two problems: emitting true when the lib wasn't loading and not working in error case :). So now the public event service is used, the error works and the `isLoading$` is still there and marked as deprecated.

**Comment by FabienDehopre** [comment]
any idea when the change from #1504 will be released?

**Comment by damienbod** [comment]
I will do this tomorrow or Saturday after testing

**Comment by FabienDehopre** [comment]
I think this issue can be closed now since the PR #1504 has been merged and released.
